mixer/adapter/stackdriver/helper/common.go

```diff
@@ -19,9 +19,15 @@ import (
 
 	gapiopts "google.golang.org/api/option"
 
+	md "cloud.google.com/go/compute/metadata"
+
 	"istio.io/istio/mixer/adapter/stackdriver/config"
+	"istio.io/istio/mixer/pkg/adapter"
 )
 
+var mdOnGCE = md.OnGCE
```

Comment:
what do you think about making this something like:

```
type shouldFillFn func (*config.Params) bool
type projectIDFn func() (string, error)

type ProjectIDFiller struct {
   shouldFill shouldFillFn
   projectID projectIDFn
}

func (p ProjectIDFiller) FillProjectID(c adapter.Config) {
  cfg := c.(*config.Params)
  if !p.shouldFill(cfg) {
    return
 }
 if pid, err := p.ProjectID(); err == nil {
    cfg.ProjectId = pid
  }
}

func NewProjectIDFiller(s shouldFillFn, p projectIDFn) {
  return &ProjectIDFiller{s, p}
}

----
stackdriver.go

type(
...
 	builder struct {
 		m metric.HandlerBuilder
 		l logentry.HandlerBuilder
                pf helper.ProjectIDFiller
 	}
...
)

func newBuilder() {
   return &builder{
       ....,
       pf: helper.NewProjectIDFiller( 
             func(c *config.Params) bool { return cfg.ProjectId == "" && md.OnGCE() }}, 
             md.ProjectID) 
   },
}

...
b.pf.FillProjectID(c)
...
```

I'm not sure it is better, but it might make things slightly clearer (with some cleaning up). Thoughts?